src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java

```diff
@@ -692,93 +690,124 @@ public void scan(HttpMessage msg, String param, String origParamValue) {
-
-							if (confirmExpressionBodyOutput[booleanStrippedUnstrippedIndex].compareTo(normalBodyOutput[booleanStrippedUnstrippedIndex]) != 0) {
-								//the confirm query did not return the same results.  This means that arbitrary queries are not all producing the same page output.
-								//this means the fact we earier reproduced the original page output with a modified parameter was not a coincidence
-
-								//Likely a SQL Injection. Raise it
-								String extraInfo = null;
-								if (strippedOutput[booleanStrippedUnstrippedIndex]) {
-									extraInfo = Constant.messages.getString(MESSAGE_PREFIX + "alert.expressionbased.extrainfo", modifiedParamValue, "");
-								} else {
-									extraInfo = Constant.messages.getString(MESSAGE_PREFIX + "alert.expressionbased.extrainfo", modifiedParamValue, "NOT ");
+							String modifiedExpressionOutputUnstripped = msg4.getResponseBody().toString();
+							String modifiedExpressionOutputStripped = this.stripOff(modifiedExpressionOutputUnstripped, modifiedParamValue);
+
+							//set up two little arrays to ease the work of checking the unstripped output, and then the stripped output
+							String normalBodyOutput[] = {mResBodyNormalUnstripped, mResBodyNormalStripped};
+							String expressionBodyOutput[] = {modifiedExpressionOutputUnstripped, modifiedExpressionOutputStripped};
+							boolean strippedOutput[] = {false, true};
+
+							for (int booleanStrippedUnstrippedIndex = 0; booleanStrippedUnstrippedIndex < 2 && !sqlInjectionFoundForUrl && countExpressionBasedRequests < doExpressionMaxRequests; booleanStrippedUnstrippedIndex++) {
+								//if the results of the modified request match the original query, we may be onto something. 
+								if (expressionBodyOutput[booleanStrippedUnstrippedIndex].compareTo(normalBodyOutput[booleanStrippedUnstrippedIndex]) == 0) {
+									if (this.debugEnabled) {
+										log.debug("Check 4, " + (strippedOutput[booleanStrippedUnstrippedIndex] ? "STRIPPED" : "UNSTRIPPED") + " html output for modified expression parameter [" + modifiedParamValue + "] matched (refreshed) original results for " + refreshedmessage.getRequestHeader().getURI());
+									}
+									//confirm that a different parameter value generates different output, to minimise false positives
+									// this time get a bigger value
+
+									String modifiedParamValueConfirm = "";
+
+									if (check4Variant[booleancheck4VariantIndex] == "plus"){
+										//and set the parameter value to a string value like "4-2", if the original parameter value was "1"
+										int paramPlusTwo = paramAsInt + 3;
+										modifiedParamValueConfirm = String.valueOf(paramPlusTwo) + "-2";
+									}
+									else if (check4Variant[booleancheck4VariantIndex] == "mult"){
+										//and set the parameter value to a string value like "4/2", if the original parameter value was "1"
+										int paramPlusTwo = paramAsInt * 4;
```

Comment:
The variable name should change, it isn't a "PlusTwo" at this point. Perhaps `increasedParamValueConfirm` or `modifiedParamValueConfirm` or something more generic like that to fit both variants?